applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c

```diff
@@ -106,6 +109,42 @@ static int next_free_audio_stream_index_get(uint8_t *index)
 	return -ENOSPC;
 }
 
+/**
+ * @brief  Get the presentation delay for al headsets
+ *
+ * @param  sel_max Selection of the use of the minimum or maximum
+ *                 preferred presentation delay (TRUE = max; FALSE = min)
+ * @param  index   The index of the hesdset to test against
+ * @param  pd      Pointer to the absolute presentation delay for the
+ *                 set of headsets
+ * @param  pref_pd Pointer to the preferred presentation delay for the
+ *                 set of headsets
+ */
+static void find_heaset_pres_delay(bool sel_max, int index, uint32_t *pd, uint32_t *pref_pd)
+{
+	if (sel_max == true) {
+		*pd = headsets[index].pref_qos.pd_max;
```

Comment:
I might be overlooking something here, but where are the values from the gateway itself? We should make sure that we don't just use the headsets values, but make sure they are within the range set by the gateway as well

Reply:
We don't need the gateway values as these are only for the headset, the return path I thought negotiated a separate presentation delay signalled by the gateway when we have an audio return path from the headset. Ah so yes in bi-directional mode I will need to test the presentation delay signalled by the gateway (headset) to the headset (gateway) when the connection is established. I need to review the code on this!

Reply:
Yes we do, even though one headset could have a minimum presentation delay of 2ms, the gateway might want to set the presentation delay to 10ms because it knows that it is sending to 4 different headsets.

Reply:
> We don't actually send a timestamp on the mic path so no negotiation of pres delay with the gateway.
I guess this was for this discussion? It's not about the mic path, it's about gateway basically saying: "All the headsets that are connected to me should have a presentation delay of 10ms" and then headset can say: "I can support presentation delays between 4ms and 40ms, but I prefer it to be between 4ms and 8ms". The gateway should then be able to have the final say and set the presentation delay to 10ms for all headsets.

The best scenario is that a presentation delay ends up within the range of both the headset and the gateways preferred range, but if it doesn't it should be the gateway that has the final say since it is the only one who knows how many headset it is connected to.

Reply:
Right I understand now thanks. So if the initiator needs a specific value, for lip sync say, it would use this if it could. In this case it would force the application to override the acceptors if it was in range. Will the application error if it can not use it's required presentation delay?